### PR TITLE
Delete Mongo recipe parameters test

### PR DESCRIPTION
**Pull request by kachawla**
# Description

Deleting Mongo recipe parameters test. This test has been disabled for a while and the scenario it covers is now being tested by https://github.com/radius-project/radius/blob/main/test/functional/shared/resources/recipe_bicep_test.go#L44

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: #issue_number

### Summary
🚫🐛🚀

Removed a flaky integration test for MongoDB recipe parameters from `test/functional/datastoresrp/resources/mongodb_test.go`. This was done to avoid failures caused by cosmosdb resource management issues.

> _`Test_MongoDB_RecipeParameters`_
> _Flaky test skipped - issue with cosmosdb_
> _Winter of testing_

### Walkthrough
*  Remove flaky test function `Test_MongoDB_RecipeParameters` from `resource_test` package ([link](https://github.com/radius-project/radius/pull/6806/files?diff=unified&w=0#diff-3537a95d4323706fe81435869ceaee11338952061d8f3c4d4c0dd526d761b440L121-L176))

### CODE REVIEW

**Comment by rynowak** [comment]
Glad we're pruning some of the places that cloud resources are used in a test without needing them 👍

File: test/functional/datastoresrp/resources/mongodb_test.go [hunk 1]
```diff
@@ -118,62 +118,6 @@ func Test_MongoDB_Recipe(t *testing.T) {
...
 }
 
-// Test_MongoDB_Recipe_Parameters validates the creation of a mongoDB from recipe with parameters passed by operator while linking recipe
-// and developer while creating the mongoDatabase resource.
-// If the same parameters are set by the developer and the operator then the developer parameters are applied in to resolve conflicts.
-// Container uses the mongoDB resource to connect to the mongoDB resource
-func Test_MongoDB_RecipeParameters(t *testing.T) {
-	t.Skip("Skipping test as creating/deleting cosmosdb resource is unreliable - https://github.com/radius-project/radius/issues/5929")
-
-	template := "testdata/datastoresrp-resources-mongodb-recipe-parameters.bicep"
```
Comment (rynowak) [inline test/functional/datastoresrp/resources/mongodb_test.go:128]: Is this file still around for us to clean up?

**Review by rynowak — DISMISSED** [review]

File: test/functional/datastoresrp/resources/mongodb_test.go (hunk 1 above)
Comment (kachawla) [reply test/functional/datastoresrp/resources/mongodb_test.go:128]: Ah yes, delete this and the recipe file.

**Review by rynowak — APPROVED** [review]

**Comment by rynowak** [comment]
Thanks for doing this 👍

**Pull request merged**